modules/javafx.web/src/test/java/test/javafx/scene/web/EventListenerLeakTest.java

```diff
@@ -627,13 +649,370 @@ public void TestStrongRefNewContentLoad() throws Exception {
+    public void oneWebViewRefCountTest() throws Exception {
+        webView2 = null; // unused
+
+        // Load HTML content and get list of DOM nodes
+        loadContent(webView1, HTML);
+
+        final List<WeakReference<MyListener>> listeners = new ArrayList<>();
+        submit(() -> {
+            domNodes1 = getDomNodes(webView1);
+            assertEquals(NUM_DOM_NODES, domNodes1.size());
+
+            // Create listeners
+            MyListener listener = MyListener.create();
+            listeners.add(new WeakReference<>(listener));
+
+            for (int i = 0; i < 3; i++) {
+                domNodes1.get(i).addEventListener("click", listeners.get(0).get(), false);
+            }
+        });
+
+        submit(() -> {
+            // Send clilck events
+            click(webView1, 0);
+            click(webView1, 1);
+            click(webView1, 2);
+        });
+
+        // Verify that the events are delivered to the listeners (0 and 2 are same)
+        Thread.sleep(100);
+        assertEquals("Click count", 3, listeners.get(0).get().getClickCount());
+
+        //save for later
+        MyListener tmpListener = listeners.get(0).get();
+
+        // remove previously registered listeners fro dom nodes
+        submit(() -> {
+            domNodes1 = getDomNodes(webView1);
+            assertEquals(NUM_DOM_NODES, domNodes1.size());
+
+            //
+            for (int i = 0; i < 3; i++) {
+                domNodes1.get(i).removeEventListener("click", listeners.get(0).get(), false);
+            }
+        });
+
+        submit(() -> {
+            // Send clilck events
+            click(webView1, 0);
+            click(webView1, 1);
+            click(webView1, 2);
+        });
+
+        // Verify that the events are delivered to the listeners (0 and 2 are same)
+        Thread.sleep(100);
+        assertEquals("Click count", 3, listeners.get(0).get().getClickCount());
+
+
+        // add events listeners again
+        submit(() -> {
+            domNodes1 = getDomNodes(webView1);
+            assertEquals(NUM_DOM_NODES, domNodes1.size());
+
+            // Create another listeners
+            MyListener listener = MyListener.create();
+            listeners.add(new WeakReference<>(listener));
+
+            for (int i = 0; i < 3; i++) {
+                domNodes1.get(i).addEventListener("click", listeners.get(1).get(), false);
+            }
+        });
+
+        tmpListener = null;
+        //save for later
+        MyListener tmpListener1 = listeners.get(0).get();
+
+        submit(() -> {
+            // Send clilck events
+            click(webView1, 0);
+            click(webView1, 1);
+            click(webView1, 2);
+        });
+
+        // Verify that the events are delivered to the listeners (0 and 2 are same)
+        Thread.sleep(100);
+        assertEquals("Click count", 6, listeners.get(1).get().getClickCount() + listeners.get(0).get().getClickCount());
+
+        // add events listeners again
```

Comment:
I think you may have misunderstood my comment. I didn't mean to suggest that you should remove the blocks of code that you removed in the most recent update -- it was performing a quite useful check. Rather I meant that since you were removing the listeners, the comment `add listeners again` was wrong and should say `remove listeners again`.